File: Study-plan-iOS/Study-plan/Models/Task.swift
import Foundation

struct Task: Identifiable, Codable, Equatable {
    var id: UUID = UUID()
    var date: Date = .now
    var timeRange: String = ""
    var subject: String = ""
    var content: String = ""
    var duration: Int = 0
    var pages: Int = 0
    var priority: Int = 0
    var status: Int = 0
    var specificTime: String = ""
    var sortIndex: Int = 0
    var calendarEventID: String?
}

struct TimerState: Codable, Equatable {
    var isRunning = false
    var isPaused = false
    var isCountdown = true
    var allowOverflow = false
    var totalSeconds = 1500
    var elapsedSeconds = 0
    var startedAt: Date?
    var pausedAt: Date?
    var activeTaskID: UUID?
}

File: Study-plan- Android/app/src/main/java/com/example/studyplan/Task.java
package com.example.studyplan;

public class Task {
    public long id;
    public String date;
    public String timeRange;
    public String subject;
    public String content;
    public int duration;
    public int pages;
    public int priority;
    public int status;
    public String specificTime;

    public Task() {}

    public Task(String date, String timeRange, String subject,
                String content, int duration, int pages, int priority) {
        this.date = date;
        this.timeRange = timeRange;
        this.subject = subject;
        this.content = content;
        this.duration = duration;
        this.pages = pages;
        this.priority = priority;
        this.status = 0;
        this.specificTime = null;
    }

    public Task(String date, String timeRange, String subject,
                String content, int duration, int pages, int priority, String specificTime) {
        this.date = date;
        this.timeRange = timeRange;
        this.subject = subject;
        this.content = content;
        this.duration = duration;
        this.pages = pages;
        this.priority = priority;
        this.status = 0;
        this.specificTime = specificTime;
    }
}
